internal/trust/remotes.go

```diff
@@ -260,11 +260,17 @@ func (r *Remotes) Count() int {
 	return len(r.data)
 }
 
+// RemotesByName returns a copy of the list of peers, keyed by each system's name.
 func (r *Remotes) RemotesByName() map[string]Remote {
 	r.updateMu.RLock()
 	defer r.updateMu.RUnlock()
 
-	return r.data
+	remoteData := make(map[string]Remote, len(r.data))
```

Comment:
Feels like we should have a mutex protecting access to r.data

Reply:
There's a mutex in all of these, it's just deferred from the top of the function. I figured it's better to block for the whole duration of the function so we don't try to read in the middle of the operation.